src/odemis/dataio/test/multi_tiff_test.py

```diff
@@ -0,0 +1,102 @@
+    test_metadata = {}
+
+    for fname in filenames:
+
+        try:
+            data = open_acquisition(fname)[0].getData()
+
+            # format metadata for tests
+            tmd = data.metadata
+            tmd["filename"] = os.path.basename(fname).replace(".tif", "")
+            tmd["shape"] = data.shape
+            tmd["dtype"] = str(data.dtype)
+            tmd["length"] = 1
+
+            # convert timestamp into datetime
+            ts = data.metadata[model.MD_ACQ_DATE]
+            tmd["timestamp"] = ts
+            tmd["datetime"] = datetime.datetime.fromtimestamp(ts).strftime('%Y-%m-%d %H:%M:%S')
+
+            test_metadata[tmd["filename"]] = tmd
+
+        except:
+            continue # ignore files that cannot be read for now
+
+    # save test_metadata as json
+    import json
+    with open(os.path.join(path, "test_metadata.json"), "w") as f:
+        json.dump(test_metadata, f)
+    
+    return test_metadata
+
+
+class TestMultiTiffIO(unittest.TestCase):
+
+  def test_manufacturer_metadata_parsers(self):
+    """Test open_data function with multiple manufacturers and metadata parsers."""
+
+    # test data path (only available on hosted runner atm)
+    TEST_PATH = os.path.join(f"{os.path.expanduser('~')}", "test_data/meteor/test_images")
+    IGNORED_KEYS = ["shape", "dtype", "length", "filename", "timestamp", "datetime"]
+
+    # if test path doesnt exist, we are on github actions, just ignore for now
+    # TODO: download test data for github actions
+    if not os.path.exists(TEST_PATH):
+        logging.warning(f"Test data not found at {TEST_PATH}. Skipping test.")
+        return
+
+    # open metadata from test_metadata.json
+    with open(os.path.join(TEST_PATH, "test_metadata.json"), "r") as f:
+        test_metadata = json.load(f)    
+
+    for fname, md in test_metadata.items():
+
+        # open data    
+        fname = os.path.join(TEST_PATH, "**", f"{fname}*.tif")
+        fname = glob.glob(fname, recursive=True)[0]
+        data = tiff.open_data(fname)
+
+        # assert data
+        self.assertIsInstance(data, tiff.AcquisitionDataTIFF)
+        self.assertEqual(len(data.content), md["length"])
+        numpy.testing.assert_array_equal(data.content[0].shape, md["shape"])
+        self.assertEqual(data.content[0].dtype.__name__, md["dtype"])
+
+        for key in md.keys():
+            if key in IGNORED_KEYS:
+                continue # skip
+            self.assertIn(key, data.content[0].metadata)
+
+        for key, value in md.items():
+            if key in IGNORED_KEYS:
+                continue # skip
+            if key in [model.MD_PIXEL_SIZE, model.MD_POS]: # array equal
+                numpy.testing.assert_array_equal(value, data.content[0].metadata[key])
+            elif key in [model.MD_DWELL_TIME]:
+                numpy.testing.assert_almost_equal(value, data.content[0].metadata[key])
+            else:
+                self.assertEqual(value, data.content[0].metadata[key])
```

Comment:
I think you can combine the two loops

```suggestion
        for key, value in md.items():
            if key in IGNORED_KEYS:
                continue # skip
            self.assertIn(key, data.content[0].metadata)
            if key in [model.MD_PIXEL_SIZE, model.MD_POS]: # array equal
                numpy.testing.assert_array_equal(value, data.content[0].metadata[key])
            elif key in [model.MD_DWELL_TIME]:
                numpy.testing.assert_almost_equal(value, data.content[0].metadata[key])
            else:
                self.assertEqual(value, data.content[0].metadata[key])
```